### PR TITLE
test: add special cases tests in `math/base/special/xlog1py`

### PR DESCRIPTION
**Pull request by ashish-066**
Resolves N/A.

## Description

This pull request adds explicit unit tests covering infinite-value edge cases for `xlog1py`.

The current implementation already exhibits well-defined behavior for `±Infinity` inputs. These tests make that behavior explict

Specifically, this PR verifies:
- Behavior when `x` is `±Infinity`
- Behavior when `y` is `Infinity`
- Interactions between zero and infinite values

## Related Issues

This pull request has no related issues.

## Questions

No.

## Other

No.

## Checklist

- [x] Read, understood, and followed the [contributing guidelines][contributing].

### AI Assistance

- [x] Yes
- [ ] No

If you answered "yes" above, how did you use AI assistance?

- [ ] Code generation (e.g., when writing an implementation or fixing a bug)
- [ ] Test/benchmark generation
- [ ] Documentation (including examples)
- [x] Research and understanding

#### Disclosure

I consulted AI only to understand the stdlib codebase, existing behavior, and testing conventions. All tests in this pull request were written, validated, and run manually.

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/math/base/special/xlog1py/test/test.js [hunk 1]
```diff
@@ -173,3 +173,26 @@ tape( 'the function evaluates `x * ln(y+1)` for large `x` and `y`', function tes
 	}
 	t.end();
 });
+tape( 'the function handles infinite values', function test( t ) {
```
Comment (Lokeshranjan8) [inline lib/node_modules/@stdlib/math/base/special/xlog1py/test/test.js:179]: Missing a blank line before this function definition. Please add one.

File: lib/node_modules/@stdlib/math/base/special/log1pmx/test/test.js [hunk 2]
```diff
@@ -88,3 +88,20 @@ tape( 'the function returns a Taylor series expansion otherwise', function test(
 	}
 	t.end();
 });
+tape( 'the function handles special values', function test( t ) {
```
Comment (Lokeshranjan8) [inline lib/node_modules/@stdlib/math/base/special/log1pmx/test/test.js]: here also

File: lib/node_modules/@stdlib/math/base/special/xlog1py/test/test.js (hunk 1 above)
Comment (ashish-066) [reply lib/node_modules/@stdlib/math/base/special/xlog1py/test/test.js:179]: @Lokeshranjan8  added

Comment (ashish-066) [reply lib/node_modules/@stdlib/math/base/special/xlog1py/test/test.js]: @Lokeshranjan8 done

File: lib/node_modules/@stdlib/math/base/special/xlog1py/test/test.js [hunk 3]
```diff
@@ -173,3 +173,27 @@ tape( 'the function evaluates `x * ln(y+1)` for large `x` and `y`', function tes
 	}
 	t.end();
 });
+
+tape( 'the function handles infinite values', function test( t ) {
+	var v;
+
+	v = xlog1py( Infinity, 1.0 );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/xlog1py/test/test.js]: Two comments:

1. All test messages should be `returns expected value`. No need for custom bespoke test messages.
2. We don't use `Infinity` and `-Infinity`. We use associated constants packages. See other tests in `math/base/special`.

**Review by kgryte — CHANGES_REQUESTED** [review]
Left initial comments.

File: lib/node_modules/@stdlib/math/base/special/xlog1py/test/test.js (hunk 3 above)
Comment (ashish-066) [reply lib/node_modules/@stdlib/math/base/special/xlog1py/test/test.js]: done changes after checking other function tests
